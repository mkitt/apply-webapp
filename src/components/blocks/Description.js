// @flow
import React, { type Node } from 'react'
import { color } from 'styled-system'
import glamorous from 'glamorous'
import { noob } from '../../lib/utils'

type Props = {
  children: Node,
  isRow?: boolean,
  term: string,
}

const Description = ({ children, isRow, term, ...props }: Props) => (
  <Dl isRow={isRow} {...props}>
    <Dt isRow={isRow} >{term}</Dt>
    <Dd>{children}</Dd>
  </Dl>
)

Description.defaultProps = {
  isRow: false,
}

export default Description

// -------------------------------------

const rowDlRule = {
  flexFlow: 'row nowrap',
}

const Dl = glamorous.dl(
  {
    display: 'flex',
    flexDirection: 'column',
    marginTop: 8,
    marginBottom: 8,
    color: 'currentColor',
  },
  props => ({
    ...(props.isRow ? rowDlRule : noob),
  }),
  color,
)

const rowDtRule = {
  minWidth: 192,
}

const Dt = glamorous.dt(
  {
    fontSize: 14,
    fontWeight: 600,
    textTransform: 'uppercase',
    color: 'currentColor',
  },
  props => ({
    ...(props.isRow ? rowDtRule : noob),
  }),
)

const Dd = glamorous.dd({
  display: 'flex',
  alignItems: 'center',
  marginLeft: 0,
  fontSize: 16,
  fontWeight: 200,
  color: 'currentColor',
})
