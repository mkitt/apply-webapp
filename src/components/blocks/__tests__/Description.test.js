import React from 'react'
import renderer from 'react-test-renderer'
import Description from '../Description'

describe('<Description />', () => {
  it('shows the correct snapshot tree for the <Description> block', () => {
    const component = (
      <Description term="Term">
        <span>Definition</span>
      </Description>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the <Description> block as a row', () => {
    const component = (
      <Description isRow term="Term Row">
        <span>Definition Row</span>
      </Description>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
