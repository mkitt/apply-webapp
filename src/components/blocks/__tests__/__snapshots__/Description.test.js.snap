// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description /> shows the correct snapshot tree for the <Description> block 1`] = `
.glamor-2,
[data-glamor-2] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  margin-top: 8px;
  margin-bottom: 8px;
  color: currentColor;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.glamor-0,
[data-glamor-0] {
  font-size: 14px;
  font-weight: 600;
  text-transform: uppercase;
  color: currentColor;
}

.glamor-1,
[data-glamor-1] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  margin-left: 0;
  font-size: 16px;
  font-weight: 200;
  color: currentColor;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

<dl
  className="glamor-2"
>
  <dt
    className="glamor-0"
  >
    Term
  </dt>
  <dd
    className="glamor-1"
  >
    <span>
      Definition
    </span>
  </dd>
</dl>
`;

exports[`<Description /> shows the correct snapshot tree for the <Description> block as a row 1`] = `
.glamor-1,
[data-glamor-1] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  margin-left: 0;
  font-size: 16px;
  font-weight: 200;
  color: currentColor;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-2,
[data-glamor-2] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  margin-top: 8px;
  margin-bottom: 8px;
  color: currentColor;
  flex-flow: row nowrap;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-flex-flow: row nowrap;
}

.glamor-0,
[data-glamor-0] {
  font-size: 14px;
  font-weight: 600;
  text-transform: uppercase;
  color: currentColor;
  min-width: 192px;
}

<dl
  className="glamor-2"
>
  <dt
    className="glamor-0"
  >
    Term Row
  </dt>
  <dd
    className="glamor-1"
  >
    <span>
      Definition Row
    </span>
  </dd>
</dl>
`;
